aruco_detector:
  ros__parameters:
    camera_module: hal_allied_vision_camera
    pose_topic_hz: 30
    marker_side: 0.1
    
    # Specify different side for each marker id. If enable False or id not in custom dict, the node 
    # will use the marker_side specified above
    custom_marker_side_dict:
      enable: True
      number_of_entries: 2
      entry_0:
        id: 69
        marker_side: 0.1
      entry_1:
        id: 0
        marker_side: 0.03

    publishers:
      marker_image: /wac/target_tracking/marker_image
      marker_transform_prefix: /wac/target_tracking/camera_to_marker_transform/marker_
      marker_presence_prefix: /wac/target_tracking/camera_to_marker_presence/marker_

    client_services:
      stop_camera: /wac_camera/get_cam_state

    subscribers:
      raw_frame: /wac_camera/raw_frame

    frames: 
      camera_link: wac_camera_link
      marker_link_prefix: wac_marker_link_

    aruco:
      dict: 5X5_100

    camera_optic_length: 8mm

    grid:
      detect_grid: False
